File: android/app/src/main/kotlin/in/cashify/androidtrc/module/engineer/api/response/ReturnPartData.kt
package `in`.cashify.androidtrc.module.engineer.api.response

import `in`.cashify.androidtrc.common.api.BaseResponse
import com.google.gson.annotations.SerializedName


/**
 * Created by Avaneesh Maurya on 31,Agosto,2019
 */
class ReturnPartData : BaseResponse() {
    @SerializedName("pbr")
    var partBarcode: String? = null
    @SerializedName("pid")
    var partId: String? = null
    @SerializedName("rr")
    var returnReason: String? = null
    @SerializedName("remark")
    var remark: String? = null
    @SerializedName("prid")
    var prid:Int? = null
}

File: lib/src/modules/engineer/my_devices/wip_devices/view_parts/part_detail/models/return_part_data.dart
import 'package:json_annotation/json_annotation.dart';
part 'return_part_data.g.dart';

@JsonSerializable()
class ReturnPartData {
  @JsonKey(name: "pbr")
  String? partBarcode;
  @JsonKey(name: "pid")
  String? partId;
  @JsonKey(name: "rr")
  String? returnReason;
  @JsonKey(name: "remark")
  String? remark;
  @JsonKey(name: "prid")
  int? prid;

  ReturnPartData(this.partBarcode, this.partId, this.returnReason, this.remark, this.prid);

  static ReturnPartData fromJson(Map<String, dynamic> json) => _$ReturnPartDataFromJson(json);
  Map<String, dynamic> toJson() => _$ReturnPartDataToJson(this);
}
